cpp/src/jit/cache.cpp

```diff
@@ -21,19 +21,28 @@
 #include <unistd.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <pwd.h>
 
 namespace cudf {
 namespace jit {
 
-// Default `LIBCUDF_KERNEL_CACHE_PATH` to `$TEMPDIR/cudf_$CUDF_VERSION`.
+// Get the directory in home to use for storing the cache
+boost::filesystem::path get_user_home_cache_dir() {
+  auto home_dir = std::getenv("HOME");
+  if (home_dir != nullptr) {
+    return boost::filesystem::path(home_dir) / ".cudf" / std::string{CUDF_STRINGIFY(CUDF_VERSION)};
+  } else {
+    return boost::filesystem::path();
+  }
+}
+
+// Default `LIBCUDF_KERNEL_CACHE_PATH` to `$HOME/.cudf/$CUDF_VERSION`.
 // This definition can be overridden at compile time by specifying a
 // `-DLIBCUDF_KERNEL_CACHE_PATH=/kernel/cache/path` CMake argument.
 // Use `boost::filesystem` for cross-platform path resolution and dir
 // creation. This path is used in the `getCacheDir()` function below.
```

Comment:
How hard would it be to make this a runtime environment variable instead of a compile time variable? For a user needing this not in their home directory that means they'd need to recompile libcudf which can often be a non-starter.

Reply:
It already supports that. 

https://github.com/rapidsai/cudf/blob/c07e46ee9304db6cbc6b63e057eb777166980dc1/cpp/src/jit/cache.cpp#L54

The compile time variable is for when runtime variable is not available. The default home directory location is used when neither are available.